### PR TITLE
now component removal doesn't generate an exception

### PR DESCRIPTION
**Pull request by RogerNogue**
bug solved

### CODE REVIEW

File: Source/GameObject.cpp [hunk 1]
```diff
@@ -236,14 +236,11 @@ std::vector<Component*> GameObject::GetComponentsInChildren(ComponentType type)
...
 {
-	for (std::vector<Component*>::iterator it = components.begin(); it != components.end(); ++it)
-	{
-		if (*it == &component)
-		{
-			(*it)->CleanUp();
-			components.erase(it);
-			RELEASE(*it);
-			return;
+	for (int i = 0; i < components.size(); ++i) {
```
Comment (sondy99) [inline Source/GameObject.cpp]: Code styling: the "{" should be down :)

Comment (AlbertVVila) [reply Source/GameObject.cpp]: Add return statement after deleting, thus you won't have to iterate after removing the component.

Comment (RogerNogue) [reply Source/GameObject.cpp]: done!

**Comment by solidajenjo** [comment]
Ok

**Pull request merged**